libs/isograph-react/src/core/garbageCollection.ts

```diff
@@ -101,9 +121,11 @@ function recordReachableIdsFromRecord(
         }
 
         for (const nextRecordId of ids) {
-          const nextRecord = store[nextRecordId];
+          const nextRecord = store[selection.concreteType]?.[nextRecordId];
           if (nextRecord != null) {
-            mutableRetainedIds.add(nextRecordId);
+            (mutableRetainedIds[selection.concreteType] ??= new Set()).add(
```

Comment:
lol let's split this into two separate lines for clarity. But TIL about `??=`, which I love and hate